### PR TITLE
feat: display virtual field "current rate"

### PR DESCRIPTION
**Pull request by HenningWendtland**

### CODE REVIEW

File: simple_subscription/simple_subscription/doctype/simple_subscription_item/simple_subscription_item.py [hunk 1]
```diff
@@ -1,8 +1,52 @@
...
+				"transaction_date": today(),
+				"price_list": party_details.selling_price_list,
+			}
+		)
+		return item_details.price_list_rate - (item_details.discount_amount or 0)
+
+	@property
+	def current_description(self):
+		item = frappe.get_doc("Item", self.item)
+		return item.description
```
Comment (barredterra) [inline simple_subscription/simple_subscription/doctype/simple_subscription_item/simple_subscription_item.py]:
```suggestion
		return frappe.db.get_value("Item", self.item, "description")
```

File: simple_subscription/simple_subscription/doctype/simple_subscription_item/simple_subscription_item.py [hunk 2]
```diff
@@ -1,8 +1,52 @@
...
+
+	@property
+	def item_name(self):
+		item = frappe.get_doc("Item", self.item)
+		return item.item_name
```
Comment (barredterra) [inline simple_subscription/simple_subscription/doctype/simple_subscription_item/simple_subscription_item.py]:
```suggestion
		return frappe.db.get_value("Item", self.item, "item_name")
```

File: simple_subscription/simple_subscription/doctype/simple_subscription_item/simple_subscription_item.py [hunk 3]
```diff
@@ -1,8 +1,52 @@
...
+				"price_list_currency": parent.currency,
+				"qty": self.qty,
+				"plc_conversion_rate": 1,
+				"conversion_rate": 1,
+				"customer": parent.customer,
+				"transaction_date": today(),
+				"price_list": party_details.selling_price_list,
+			}
+		)
+		return item_details.price_list_rate - (item_details.discount_amount or 0)
```
Comment (barredterra) [inline simple_subscription/simple_subscription/doctype/simple_subscription_item/simple_subscription_item.py]:
```suggestion
		parent = frappe.get_doc("Simple Subscription", self.parent)
		currency = parent.currency or frappe.get_cached_value(
			"Company", parent.company, "default_currency"
		)

		party_details = get_party_details(
			party=parent.customer,
			account=None,
			party_type="Customer",
			company=parent.company,
			posting_date=today(),
			currency=currency,
			doctype="Sales Invoice",
			fetch_payment_terms_template=False,
		)

		price_list = party_details.selling_price_list or frappe.db.get_value(
			"Price List", {"selling": 1, "currency": currency, "enabled": 1}
		)

		item_details = get_item_details(
			{
				"item_code": self.item,
				"company": parent.company,
				"doctype": "Sales Invoice",
				"currency": currency,
				"price_list_currency": currency,
				"qty": self.qty,
				"plc_conversion_rate": 1,
				"conversion_rate": 1,
				"customer": parent.customer,
				"transaction_date": today(),
				"price_list": price_list,
			}
		)
		return item_details.price_list_rate - (item_details.discount_amount or 0)
```

We need to fetch the default currency, if the current subscription doesn't have one yet. And the default price list, if there's no specific one for the customer.

**Review by barredterra — CHANGES_REQUESTED** [review]

**Comment by HenningWendtland** [comment]
@barredterra 
Can this be merged?

**Review by barredterra — APPROVED** [review]

**Pull request merged**